### ISSUE TITLE
Cannot do cabal 3.0 build for 8.6.5

### ISSUE THREAD

**Issue by jarcane** [haskell/haskell-ide-engine#1250]
Trying to install `cabal-hie-8.6.5` gives me the following error instead:

```
Warning: Failed to build documentation for haddock-api-2.22.0 (which is
required by test:wrapper-test from haskell-ide-engine-0.9.0.0, test:unit-test
from haskell-ide-engine-0.9.0.0 and others).
# stack (for cabal-hie-8.6.5)
# cabal (for cabal-hie-8.6.5)
cabal: unrecognized 'new-install' option
`--symlink-bindir=/home/annaia/.local/bin'
Error when running Shake build system:
  at src/Development/Shake/Internal/Args.hs:81:69-79:
* Depends on: cabal-hie-8.6.5
* Raised the exception:
user error (Development.Shake.command_, system command failed
Command: cabal new-install --write-ghc-environment-files=never --symlink-bindir=/home/annaia/.local/bin exe:hie --overwrite-policy=always
Exit code: 1
Stderr:
cabal: unrecognized 'new-install' option
`--symlink-bindir=/home/annaia/.local/bin'
)
```

I have ghc-8.6.5 and cabal 3.0 installed locally so I'm presuming this is because somehow the stack temp cabal isn't the right version?

Anyway to fix this or is there something I'm missing?

**Comment by alanz** [comment]
Has cabal 3.0 been released yet?

**Comment by nponeccop** [comment]
> Has cabal 3.0 been released yet?

No. However:

- We do build against GHC-nightly
- We'll need to support it when it's out
- Having a version check is a good idea to prevent breakage. We do check `stack` for example.

**Comment by alanz** [comment]
@nponeccop we build against *stackage* nightly, not ghc nightly.

And we have a number of dependencies, that are intricately tied to the environment. So I would not expect a new cabal to just work, without updates to `cabal-helper` and `ghc-mod` at the very least.

**Comment by nponeccop** [comment]
> @nponeccop we build against stackage nightly, not ghc nightly.
oh, my bad. I saw ghc-nightly in .circleci and thought it's against GHC HEAD built by someone else (which is not a bad idea)

**Comment by alanz** [comment]
I think this should be closed, as wontfix?

**Comment by fendor** [comment]
I think this should be addressed soonish. Would be the fix to just use `cabal v2-build`, to ensure backwards and forwards compatibility?

**Comment by mpickering** [comment]
I think using `v2-build` is an easy way to fix this?

**Comment by fendor** [comment]
Actually, the problem seems to be:
```
cabal: unrecognized 'new-install' option `--symlink-bindir=/home/annaia/.local/bin'
```

**Comment by fendor** [comment]
This problem is becoming more urgent. The fix, per se, is simple, just check for the version string, before using `v2-install`. 
However, the question arises what version range we aim to support.

**Comment by jneira** [comment]
Note that cabal-install-3.0.0.0 had been shiped with `--install-method=symlink` and it seems [it will be fixed with `3.0.0.1`](https://github.com/haskell/cabal/issues/5837#issuecomment-525118128).
So maybe in windows we should explicitly set that flag if we ant support `3.0.0.0`.

**Comment by NickSeagull** [comment]
Perhaps this issue can be unpinned now?

**Comment by wilx** [comment]
This is a real bummer. Can we get the 3.0.0.1 soon?

**Comment by jneira** [comment]
@wilx the`install.script` already support cabal-3.0.0.0`, setting `--install-method=copy` auto for windows
